Common/Sources/View/Error/ErrorView.swift

```diff
@@ -17,21 +17,36 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 import HyperwalletSDK
+import Insights
 import UIKit
 
 /// The class to handle UI errors
 public final class ErrorView {
-    weak var viewController: UIViewController!
-    var error: HyperwalletErrorType
+    private let errorTypeConnection = "CONNECTION"
+
+    private weak var viewController: UIViewController!
+    private var error: HyperwalletErrorType
+    private var pageName: String
+    private var pageGroup: String
+    private let hyperwalletInsights: HyperwalletInsightsProtocol
 
     /// Initializer to initialize the class with errors to be displayed and the viewcontroller responsible
     /// to display the errors
     /// - Parameters:
     ///   - viewController: view controller that contains errors
     ///   - error: hyperwallet error
-    public init(viewController: UIViewController, error: HyperwalletErrorType) {
+    ///   - pageName: The Page or screen that is currently visible
+    ///   - pageGroup: The group of the Page or screen that is currently visible
+    public init(viewController: UIViewController,
+                hyperwalletInsights: HyperwalletInsightsProtocol,
```

Comment:
default it to hyperwalletInsights: HyperwalletInsightsProtocol = HyperwalletInsights so that you do not have to change all the modules to pass this value. But you can still change the test classes to pass mock object

Reply:
Our current architecture provides us to inject the HyperwalletInsights instance via the Presenter class initializer only. This default value is set in the View's method ShowError(...).

Reply:
I think, instead of adding it in the protocol and init method for the presenter - we should just add it to the default initilizer of `ErrorView` as Jaspreet is suggesting - we can remove it from the signature of `showError`

Reply:
fixed to default